lib/class-wp-rest-block-editor-settings-controller.php

```diff
@@ -0,0 +1,260 @@
+<?php
+/**
+ * REST API: WP_REST_Block_Editor_Settings_Controller class
+ *
+ * @package    WordPress
+ * @subpackage REST_API
+ */
+
+/**
+ * Core class used to retrieve the block editor settings via the REST API.
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Block_Editor_Settings_Controller extends WP_REST_Controller {
+	/**
+	 * Constructs the controller.
+	 */
+	public function __construct() {
+		$this->namespace = '__experimental/wp-block-editor/v1';
+		$this->rest_base = 'settings';
+	}
+
+	/**
+	 * Registers the necessary REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks whether a given request has permission to read block editor settings
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|bool True if the request has permission, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) {// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			$error = __( 'Sorry, you are not allowed to read the block editor settings.', 'gutenberg' );
+			return new WP_Error( 'rest_cannot_read_settings', $error, array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns the block editor's settings
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$settings = array_merge(
+			apply_filters( 'block_editor_settings', array() ),
```

Comment:
This is most likely to change after this https://github.com/WordPress/gutenberg/issues/30082#issuecomment-816679803

Reply:
I expect that `gutenberg_get_block_editor_settings` is going to be the primary way to get the settings. The question is how to pass the context - the name of the editor. It looks like setting are very different for the site editor, post editor or the widgets editor.

Reply:
Something like [this](https://github.com/WordPress/gutenberg/pull/29969/files#diff-593ccf28f1034efadc57f8992c879282300de88d6672c9ba4ab5f7764a307ad5R81)? Getting the `context` from a param and passing it to `gutenberg_get_block_editor_settings`?

One question by the way, after updating with `trunk` I'm getting:

```
 Undefined index: imageSizes in /lib/editor-settings.php on line 22
```

From this [changes](https://github.com/WordPress/gutenberg/pull/30955/files), so now I'm wondering if I need to pass something else or there's something I'm missing 😅